### DISCUSSION TITLE
Feature request: Skip importing elements based on value in feed.

### DISCUSSION THREAD

**Discussion by metamachineco** [engram-design/FeedMe#1011] [Ideas]
### Description

I would like to skip certain elements from importing based on a value of a field in the feed itself.

For example: I'm building a Commerce site that imports products that are managed in an external system.
That external system contains thousands of products, but the commerce site only needs a small subset of those products.

A possible solution would be if we were able to set a field and value from the feed that FeedMe would check before importing an element.

Something like this:
```
<?php

return [
    '*' => [
        'skipElement' => [
            'skipElementField' => 'active',
            'skipElementFieldValue' => 0,
        ],
    ]
];
```

**Comment by vnali** [comment]
if writing a simple plugin is an option, you could use feed-me [built in events](https://docs.craftcms.com/feed-me/v4/developers/events.html#feeds-related-events), something like this;
```
Event::on(Process::class, Process::EVENT_STEP_BEFORE_ELEMENT_MATCH, function(FeedProcessEvent $event) {
// use craft::dd($event) to see attributes
if (..){ 
$event->isValid=false; // cancel importing with this 
}
}
```
also there are some good examples available on craftcms.stackexchange.com

**Comment by kristiansp** [comment]
Agree with @vnali that events are probably the best route (you could also make a module, which much the same, but a tiny bit easier than a plugin). Setting up the scaffolding from https://pluginfactory.io/ should get you far along the way.
